lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -965,6 +965,86 @@ class MongoClientInterface {
         ], cb);
     }
 
+    /**
+     * gets object metadata for a list of objects
+     * @param {String} bucketName bucket name
+     * @param {Array} objects array of objects
+     * @param {Object} log logger
+     * @param {Function} callback callback
+     * @return {undefined}
+     */
+    getObjects(bucketName, objects, log, callback) {
+        const c = this.getCollection(bucketName);
+        let vFormat = null;
+        if (!Array.isArray(objects)) {
+            return callback(errors.InternalError.customizeDescription('objects must be an array'));
+        }
+        // Function to process each document
+        const processDoc = (doc, objName, params, key, cb) => {
+            const versionIdValue = params && params.versionId ? params.versionId : undefined;
+            if (!doc && versionIdValue) {
+                return cb(null, {
+                    doc: null,
+                    versionId: versionIdValue,
+                    key,
+                });
+            }
+            // If no master found then ob ject is either non existent
+            // or last version is delete marker
+            if (!doc || doc.value.isPHD) {
+                return this.getLatestVersion(c, objName, vFormat, log, (err, _doc) => {
+                    if (err && !err.is.NoSuchKey) {
+                        return cb(err);
```

Comment:
Thanks for the comment, I have two questions around this:

* When getting all objects at once with batching, we can "only" handle the NoSuchKeys right? As any other error preventing the DB call to complete will result in the `catch` being called.
* If I understand correctly, you would like to have, instead of a `null` document (or more precisely, in addition to the `null` document), an `err`-like field for each of the returned object with the appropriate error, right? If yes, then we can return one error per call to `getLatestVersion` and let the upper code handle these.

Reply:
Here is my commit with the change: https://github.com/scality/Arsenal/pull/2134/commits/34b8ddbbe8b8515e31ea7eb7705c8988c9a786fa